sled-agent/src/bootstrap/rack_ops.rs

```diff
@@ -75,7 +76,15 @@ impl RssAccess {
                     }
                     Err(TryRecvError::Empty) => {
                         // Initialization task is still running
-                        RackOperationStatus::Initializing { id }
+                        // Check the step channel, and if it is different
+                        // than what we have, update what we have.
+                        // Ignore errors
+                        if let Ok(cur_step) = step_rx.try_recv() {
```

Comment:
Two thoughts:

* Should this be a `while let Ok(_) = ...` in case there have been multiple steps taken since the last time we were called?
* If yes, should we be using a [watch](https://docs.rs/tokio/latest/tokio/sync/watch/index.html) channel instead of an `mpsc`? We don't need the "multiple receivers" support from `watch`, but its API is more designed for something like this, I think: the task performing RSS can update it as much as it wants (infallibly even, if using `send_modify()`), and at any point the "receiving" side can ask what the current value is via `borrow()` (without having to `.await`, which is also good in this context).

Reply:
yes -> 
  I'm testing a watch version now and if that works I'll push it up.